tests/test_add_new_openwrap_partner.py

```diff
@@ -118,6 +167,220 @@ def test_missing_adpod_placement_size_1(self, mock_dfp_client):
+    settings.PREBID_BIDDER_CODE= int(1)
+    with self.assertRaises(BadSettingException):
+      tasks.add_new_openwrap_partner.main() 
+
+  @patch('settings.DFP_SAME_ADV_EXCEPTION',1,create=True)
+  def test_wrong_same_adv_exception_setting(self,mock_dfp_client):
+    """
+    It throws an exception with a Bad setting.
+    """
+    with self.assertRaises(BadSettingException):
+      tasks.add_new_openwrap_partner.main() 
+
+  @patch('settings.DFP_DEVICE_CATEGORIES',1,create=True)
+  def test_wrong_device_catagories_setting(self,mock_dfp_client):
+    """
+    It throws an exception with a Bad setting.
+    """
+    with self.assertRaises(BadSettingException):
+      tasks.add_new_openwrap_partner.main() 
+
+  @patch('settings.DFP_ROADBLOCK_TYPE','test',create=True)
+  def test_wrong_roadbloack_type_setting(self,mock_dfp_client):
+    """
+    It throws an exception with a Bad setting.
+    """
+    with self.assertRaises(BadSettingException):
+      tasks.add_new_openwrap_partner.main()  
+  
+  @patch('settings.LINE_ITEM_PREFIX',1,create=True)
+  def test_wrong_lineitem_prefix_setting(self,mock_dfp_client):
+    """
+    It throws an exception with a Bad setting.
+    """
+    with self.assertRaises(BadSettingException):
+      tasks.add_new_openwrap_partner.main() 
+
+  @patch('settings.OPENWRAP_USE_1x1_CREATIVE',1,create=True)
+  def test_wrong_use_1x1_creative_setting(self,mock_dfp_client):
+    """
+    It throws an exception with a Bad setting.
+    """
+    with self.assertRaises(BadSettingException):
+      tasks.add_new_openwrap_partner.main() 
+
+  @patch('settings.OPENWRAP_CUSTOM_TARGETING',1,create=True)
+  def test_wrong_custom_targeting_setting(self,mock_dfp_client):
+    """
+    It throws an exception with a Bad setting.
+    """
+    with self.assertRaises(BadSettingException):
+      tasks.add_new_openwrap_partner.main()
+
+  @patch('settings.OPENWRAP_CUSTOM_TARGETING',[("a", "IS", (1,2,3),1), ("b", "IS_NOT", ("4", "5", "6"))],create=True)
+  def test_wrong_custom_targeting_length_setting(self,mock_dfp_client):
+    """
+    It throws an exception with a Bad setting.
+    """
+    with self.assertRaises(BadSettingException):
+      tasks.add_new_openwrap_partner.main()
+
+  @patch('settings.OPENWRAP_CUSTOM_TARGETING',[("a", "TEST", ("1","2","3")), ("b", "TEST1", ("4", "5", "6"))],create=True)
+  def test_wrong_custom_targeting_IS_and_IS_NOT(self,mock_dfp_client):
+    """
+    It throws an exception with a Bad setting.
+    """
+    with self.assertRaises(BadSettingException):
+      tasks.add_new_openwrap_partner.main() 
+
+
+  @patch('settings.OPENWRAP_CUSTOM_TARGETING',[("a", "IS", (1)), ("b", "IS_NOT", (4))],create=True)
+  def test_wrong_custom_targeting_instance(self,mock_dfp_client):
+    """
+    It throws an exception with a Bad setting.
+    """
+    with self.assertRaises(BadSettingException):
+      tasks.add_new_openwrap_partner.main()
+  
+
+  def test_validate_CSVValues_can_not_negative(self,mock_dfp_client):
```

Comment:
Rename to test_validate_CSVValues_Negative

Reply:
Resolved